x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -56,15 +56,15 @@ export function getTransactionType({
     return urlParams.transactionType;
   }
 
-  if (!agentName || transactionTypes.length === 0) {
-    return;
-  }
-
   // The default transaction type is "page-load" for RUM agents and "request" for all others
   const defaultTransactionType = isRumAgentName(agentName)
     ? TRANSACTION_PAGE_LOAD
     : TRANSACTION_REQUEST;
 
+  if (!agentName || transactionTypes.length === 0) {
+    return defaultTransactionType;
```

Comment:
I don't think it is a problem to return the default transaction type when agent or transactionTypes is not defined. Let me know if otherwise.

Reply:
What happens when the transaction types are not loaded yet, but will come in later? Will this function return different transaction types before and after they have fully loaded?

Reply:
this might be the case. I check the status of the transaction type fetch and only return the defaultTransactionType when status equals SUCCESS and `transactionTypes.length === 0`. WDYT?

Reply:
That would be an improvement. But why do we need to return a transaction type?

Reply:
Because when we select a date in the future, which does not have any data,  the API `GET /api/apm/services/{serviceName}/transaction_types` returns `[]`, which means that transactionType is undefined. 

So in that case because of the condition that returns null when service name or transaction type is not defined we should a blank page. https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/public/components/app/transaction_overview/index.tsx#L104-L106

Reply:
I'm doing like we talked when transaction type is undefined skip the UI filter API call.